src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -142,14 +148,16 @@ public Storage storageForBucket(GoogleBucketResource bucketResource) {
   public void writeStreamToGcsFile(String path, Stream<String> contentsToWrite, String projectId) {
     logger.info("Writing contents to {}", path);
     Storage storage = gcsProjectFactory.getStorage(projectId);
-    var blobbyBoi = getBlobFromGsPath(storage, path, projectId);
-
-    try (var writer = blobbyBoi.writer(Storage.BlobWriteOption.userProject(projectId))) {
-      contentsToWrite.forEach(
+    var blob = getBlobFromGsPath(storage, path, projectId);
+    var newLineByteBuffer = ByteBuffer.wrap("\n".getBytes(StandardCharsets.UTF_8));
+    try (var writer = blob.writer(Storage.BlobWriteOption.userProject(projectId));
+        var contents = contentsToWrite) {
```

Comment:
Why did you change this from using `contentsToWrite` directly?

Reply:
I think the intent here is to make sure that `contentsToWrite` is closed at the end of the `try` block. I believe you don't need `var` here and can get the same effect by referencing the variable directly.
```suggestion
        contentsToWrite) {
```

See https://docs.oracle.com/en/java/javase/16/language/java-language-changes.html#GUID-A920DB06-0FD1-4F9C-8A9A-15FC979D5DA3

Although, I don't know if this is enough. What if `gcsProjectFactory.getStorage()` throws? If any code that comes before the `try` throws, `contentsToWrite` won't be closed. I think you'd have to put the `try` at the point where it first comes into scope to properly clean up.